internal/controller/postgrescluster/instance.go

```diff
@@ -1695,12 +1698,23 @@ func (r *Reconciler) prepareForUpgrade(ctx context.Context,
 			ConditionPGUpgradeCompleted)
 	}
 
+	var repoHostReady bool
+	repoHostReadyCondition := meta.FindStatusCondition(cluster.Status.Conditions,
+		ConditionRepoHostReady)
+	if repoHostReadyCondition != nil && repoHostReadyCondition.Status == metav1.ConditionTrue {
+		repoHostReady = true
+	}
```

Comment:
⛏️ There's a function for this, `IsStatusConditionTrue`

```suggestion
	repoHostReady := meta.IsStatusConditionTrue(cluster.Status.Conditions,
		ConditionRepoHostReady)
```

Reply:
:exploding_head: Thanks for the tip!  I know this helper can be leveraged elsewhere as well.